bin/codemods/run.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+const child_process = require( 'child_process' );
+
+/**
+ * Internal dependencies
+ */
+const config = require( './src/config' );
+
+function getLocalCodemodFileNames() {
+	// Returns all JS files in bin/codemods/src folder, except for config and helpers.
+
+	return fs.readdirSync( './bin/codemods/src' )
+		.filter( name => name !== 'config.js' && name != 'helpers.js' ) // exclude utility files
+		.filter( name => name[ 0 ] !== '.' ) // exclude dot files
+		.filter( name => path.extname( name ) === '.js' ) // only include JS files
+		.map( name => name.replace( '.js', '' ) ); // strip extension from filename
+}
+
+function getValidCodemodNames() {
+	return [
+		...getLocalCodemodFileNames(),
+		...Object.getOwnPropertyNames( config.codemodArgs ),
+	].map( name => '- ' + name ).sort();
+}
+
+function printableValidCodemodNames() {
+	return getValidCodemodNames().join('\n');
+}
+
+function generateBinArgs( name ) {
+	if ( config.codemodArgs.hasOwnProperty( name ) ) {
+		// Is the codemod defined in the codemodArgs object?
+		return config.codemodArgs[ name ];
+	}
+
+	if ( getLocalCodemodFileNames().includes( name ) ) {
+		// Is the codemod a local script defined in bin/codemods/src folder?
+		return [
+			`--transform=bin/codemods/src/${ name }.js`,
+		];
+	}
+
+	throw new Error(
+		`"${ name }" is an unrecognized codemod.`
+	);
+}
+
+function runCodemod() {
+	const args = process.argv.slice( 2 );
+	if ( args.length === 0 || args.length === 1 ) {
+		process.stdout.write( '\n' );
+		process.stdout.write( './bin/codemods/run.js [transformation name] [target(s)]\n' );
+		process.stdout.write( '\n' );
+		process.stdout.write( `Valid transformation names: \n${ printableValidCodemodNames() }\n` );
+		process.stdout.write( '\n' );
+		process.stdout.write( 'Example: "./bin/codemods/run.js commonjs-imports client/blocks client/devdocs"\n' );
+		process.stdout.write( '\n' );
+		process.exit( 0 );
+	}
+
+	const [ name, ...targets ] = args;
+	const binArgs = [
+		// jscodeshift options
+		...config.jscodeshiftArgs,
+
+		...generateBinArgs( name ),
+
+		// Transform target
+		...targets,
+	];
```

Comment:
personal style preference only, but I think having fewer lines and comments on the same line makes it easier for me to read this…

```js
const binArgs = [
	...config.jscodesiftArgs, // actually is the comment helping here at all?
	...generateBinArgs( name ),
	...targets, // transform target
];
```